packages/core/src/http-client/csrf-token-header.ts

```diff
@@ -67,14 +74,29 @@ function makeCsrfRequest<T extends HttpRequestConfig>(
       return executeHttpRequest(destination, removeSlash(axiosConfig))
         .then(response => response.headers)
         .catch(e2 => {
-          throw new ErrorWithCause(
-            'Also second try to fetch SRF token failed - No CSRF token fetched.',
-            e2
+          if (hasCsrfToken(e2)) {
+            return e2.response.headers;
+          }
+          logger.warn(
+            new ErrorWithCause(
+              'Also second try to fetch CSRF token failed - No CSRF token fetched.',
+              e2
+            )
           );
+          // todo suggest to disable csrf token handling when the api is implemented
+          return {};
         });
     });
 }
 
+function hasCsrfToken(e1): boolean {
+  return (
+    e1.isAxiosError &&
+    e1.response?.headers &&
+    e1.response.headers['x-csrf-token']
```

Comment:
Independent of the conversation above, you just need one line:
```suggestion
    e1.response?.headers?.['x-csrf-token']
```